packages/plugin-core/src/services/TextDocumentServiceFactory.ts

```diff
@@ -1,21 +1,18 @@
-import { IDendronExtension } from "../dendronExtensionInterface";
 import { TextDocumentService } from "./node/TextDocumentService";
-import { workspace } from "vscode";
 import { ITextDocumentService } from "./ITextDocumentService";
+import { container } from "tsyringe";
 
 export class TextDocumentServiceFactory {
   private static _textDocumentService: ITextDocumentService | undefined;
 
   /**
    * Instantiate TextDocumentService to be used in _extension.ts/workspace.ts
    */
-  public static create(extension: IDendronExtension): ITextDocumentService {
+  public static create(): ITextDocumentService {
```

Comment:
Now that `TextDocumentService` is injectable, we don't need or want to have separate Factory classes or methods anymore - tsyringe and its containers should act as the factory. It can also manage the singleton nature of the object. 

Let's go ahead and do the following: 
1. Delete this `TextDocumentServiceFactory` class
2. Use `container.resolve()` where this Factory is being called right now (just `workspaceActivator.ts`)
3. Define the singleton behavior inside the setupContainer() function (I left another comment there on how to do that)